setcode action

JaskRendix opened the pull request:
PR addresses the creation of the set_code action (already structured for #1385) as well as adds three msgid PO.

(I have the other version with params, so it's not a problem to reverse)

Reason? Modders, modders can define a code (numeric or alphabetical or mixed) to access, open a door, receive or trigger something.

It's composed by three parameters:
- code (the actual code);
- characters (characters of the code; eg. code is "1234", then 4 characters;
- variable: the variable that if the code is correct will be set on "on", if not will be set on "off";

eg: **property name="act2" value="set_code lepeletier,10,revolution"**

The variable will help modders with the following actions (a scenario if it's correct, another if it's wrong); above the player has to guess "rockitten". It'll be created the variable prize, so if it's correct the value will be "on", otherwise "off". 

Question: is it ok case sensible? Too complicated? Do we need all upper or lowercase? Let me know. At the end the modders can play on the difficulty of the word too (or sequence of numbers or both).

The input:
![Screenshot from 2022-11-06 08-14-34](https://user-images.githubusercontent.com/64643719/200158887-bf2cb88d-a576-4cc9-aa9e-8b39f919ad24.png)
The message, if it's correct, then it'll be correct.
![Screenshot from 2022-11-06 08-14-47](https://user-images.githubusercontent.com/64643719/200158903-fa95b0ce-dffd-42e4-b9ef-931caefc85e0.png)

kerizane commented on tuxemon/event/actions/set_code.py:
```diff
@@ -0,0 +1,65 @@
+class SetCodeAction(EventAction):
+    """
+    Set a code and checks if it's correct or not.
+    Case Sensible:
```
Sensitive, not Sensible

JaskRendix replied:
Thanks

kerizane commented on tuxemon/event/actions/set_code.py:
```diff
@@ -0,0 +1,65 @@
+
+    Script parameters:
+        code: The actual code the player needs to know.
+        characters: The number of characters/digits.
```
We could calculate this automatically by counting the number of characters in code. Or did you want to allow values longer than the actual code?

JaskRendix replied:
Automatically by counting the number of characters is perfect, so we can remove a parameter, that's perfect.

kerizane commented on tuxemon/event/actions/set_code.py:
```diff
@@ -0,0 +1,65 @@
+            open_dialog(
+                self.session,
+                [T.format("right_code", {"code": name})],
+            )
```
I don't think we should have the dialogue hard-coded. Just let the modder trigger the dialogue manually using the output variable.

JaskRendix replied:
It makes sense, so in the following steps the modder can check the variable by existing actions/conditions.

JaskRendix commented:
Removed right and wrong_code too. Modders will think about the right options;
Removed input_code hardcoded, now modders can set up;
Added instruction to add the input_code inside the PO file (translation);
Renamed parameters question and answer to be more intuitive;
Renamed on/off as right/wrong inside the variables;

kerizane commented on tuxemon/event/actions/set_code.py:
```diff
@@ -0,0 +1,54 @@
+        self.session.client.push_state(
+            state_name=InputMenu,
+            prompt=T.translate("input_code"),
+            callback=self.check_setcode,
```
We should probably make the prompt a parameter. I.e. don't hardcode "input_code" here.

JaskRendix replied:
No problem, added some instruction too, so they can immediately define the msgid.

JaskRendix committed:
set code
